src/opm/io/eclipse/EclFile.cpp

```diff
@@ -342,129 +330,110 @@ void readFormattedHeader(std::fstream& fileH, std::string& arrName,
-        int p1 = line.find_first_of("'");
+    for (int i=0; i< size; i++) {
+        p1 = file_str.find_first_of('\'',p1);
+        p2 = file_str.find_first_of('\'', p1+1);
 
-        if (p1 == -1) {
-            std::string message="Reading formatted char array, all strings must be enclosed by apostrophe (')";
+        std::string value = file_str.substr(p1 + 1, p2 - p1 - 1);
+
+        if (value.size() != 8) {
+            std::string message="Reading formatted char array, all strings should have 8 characters";
             OPM_THROW(std::runtime_error, message);
         }
 
-        while (p1 > -1) {
-            int p2 = line.find_first_of("'", p1 + 1);
-
-            if (p2 == -1) {
-                std::string message="Reading formatted char array, all strings must be enclosed by apostrophe (')";
-                OPM_THROW(std::runtime_error, message);
-            }
-
-            std::string value = line.substr(p1 + 1, p2 - p1 - 1);
-
-            if (value.size() != 8) {
-                std::string message="Reading formatted char array, all strings should have 8 characters";
-                OPM_THROW(std::runtime_error, message);
-            }
-
-            if (value == "        ") {
-                arr.push_back("");
-            } else {
-                arr.push_back(Opm::EclIO::trimr(value));
-            }
-
-            num++;
-
-            p1 = line.find_first_of("'", p2 + 1);
+        if (value == "        ") {
+            arr.push_back("");
+        } else {
+            arr.push_back(Opm::EclIO::trimr(value));
         }
+
+        p1 = p2+1;
     }
 
     return arr;
 }
 
 
-std::vector<float> readFormattedRealArray(std::fstream &fileH, const int size)
+std::vector<float> readFormattedRealArray(const std::string& file_str, const int size, long int fromPos)
 {
+
     std::function<float(const std::string&)> f = [](const std::string& val)
                                                  {
                                                      // tskille: temporary fix, need to be discussed. OPM flow writes numbers
                                                      // that are outside valid range for float, and function stof will fail
                                                      double dtmpv = std::stod(val);
                                                      return dtmpv;
                                                  };
-    return readFormattedArray<float>(fileH, size, f);
+
+    return readFormattedArray<float>(file_str, size, fromPos, f);
 }
 
 
-std::vector<bool> readFormattedLogiArray(std::fstream& fileH, const int size)
+std::vector<bool> readFormattedLogiArray(const std::string& file_str, const int size, long int fromPos)
 {
+
     std::function<bool(const std::string&)> f = [](const std::string& val)
                                                 {
-                                                    if (val == "T") {
+                                                    if (val[0] == 'T') {
```

Comment:
Here you have changed the semantics slightly - now e.g. the strings `TULL` and `TØYS` will both evaluate to `true` - is that intentional?

Reply:
This is an optimization on performance. An alternative will be to replace all line shifts characters ('\n') to space (' ') 
```
std::replace(fileStr.begin(), fileStr.end(), '\n', ' ');
```
just after reading the data from disk and before sending the strings to loadFormattedArray(const std::string& ...). It is then possible to check for the full string instead of only the first character. 

However this result in about +10% runtime. I believe that we should keep this optimization.